### ISSUE TITLE
Allow use of system/user installed Minimap2/KMC/etc

### ISSUE THREAD

**Issue by peterk87** [nlapier2/Metalign#16]
Right now Minimap2 needs to be installed in a path relative to the calling script:

https://github.com/nlapier2/Metalign/blob/94b6820da1396dd3187dc96cdb9fbb0511636d91/map_and_profile.py#L427

It's fairly easy to install Minimap2 and KMC3 with Conda from the Bioconda channel:

```bash
conda install -c bioconda minimap2 kmc -y
```

Code could be changed to:

```python
# default to using $PATH accessible minimap2
parser.add_argument('--minimap2-bin', default='minimap2')

# check the valid version of minimap2 is being used
# maybe raise exception if no valid minimap2 installed or specified?
check_minimap2(args.minimap2_bin)

# change call slightly
mapper = subprocess.Popen([minimap2_bin, '-ax', ...
```

**Comment by dkoslicki** [comment]
Good idea! We welcome a pull request, otherwise we will try to get to it after we fix a few other issues (namely #12 and CMash installation issues).

**Comment by peterk87** [comment]
I could give it a try :)

What version of Python are you targeting? Python 3.6 and later? f-strings, in-line type hints are very nice.

**Comment by dkoslicki** [comment]
@peterk87 Yes, Python >=3.6 is the goal. And goal is to get Travis CI up in the next month, so we can auto test PR's, but until then, we can just make/run manual tests.

**Comment by peterk87** [comment]
@dkoslicki  That's great to hear! Hope setting up CI isn't too tricky for you guys! I've found it difficult to get even tiny bioinformatics analyses to run in a reasonable amount of time with very limited memory. 

I've found the [cookiecutter](https://cookiecutter.readthedocs.io/en/latest/README.html#available-cookiecutters) recipe for [Python packages](https://github.com/audreyr/cookiecutter-pypackage) to have a nice Travis-CI setup (and generally nice structure for a Python package)

Recent Python tools I've been developed have been bootstrapped with: 

```bash
$ cookiecutter https://github.com/audreyr/cookiecutter-pypackage.git
```

Also has some config for auto deployment to PyPI on tagged release creation in Github.

Hope that helps!
